src/ucp/wireup/select.c

```diff
@@ -64,14 +75,15 @@ static const char *ucp_wireup_iface_flags[] = {
     [ucs_ilog2(UCT_IFACE_FLAG_GET_SHORT)]        = "get short",
     [ucs_ilog2(UCT_IFACE_FLAG_GET_BCOPY)]        = "get bcopy",
     [ucs_ilog2(UCT_IFACE_FLAG_GET_ZCOPY)]        = "get zcopy",
-    [ucs_ilog2(UCT_IFACE_FLAG_ATOMIC_ADD32)]     = "32-bit atomic add",
-    [ucs_ilog2(UCT_IFACE_FLAG_ATOMIC_ADD64)]     = "64-bit atomic add",
-    [ucs_ilog2(UCT_IFACE_FLAG_ATOMIC_FADD32)]    = "32-bit atomic fetch-add",
-    [ucs_ilog2(UCT_IFACE_FLAG_ATOMIC_FADD64)]    = "64-bit atomic fetch-add",
-    [ucs_ilog2(UCT_IFACE_FLAG_ATOMIC_SWAP32)]    = "32-bit atomic swap",
-    [ucs_ilog2(UCT_IFACE_FLAG_ATOMIC_SWAP64)]    = "64-bit atomic swap",
-    [ucs_ilog2(UCT_IFACE_FLAG_ATOMIC_CSWAP32)]   = "32-bit atomic compare-swap",
-    [ucs_ilog2(UCT_IFACE_FLAG_ATOMIC_CSWAP64)]   = "64-bit atomic compare-swap",
+    /* TODO: remove deprecated flags */
+    [ucs_ilog2(UCT_IFACE_FLAG_ATOMIC_ADD32)]     = "32-bit atomic add (deprecated)",
```

Comment:
can we remove these from UCP altogether and use only the new atomic bits?

Reply:
yes, I was going to remove it in next PRs where deprecated stuff is cleaned
for now it could be useful in debug to catch deprecated API used

Reply:
maybe we can extend/replace the function ucp_wireup_check_flags() to ucp_wireup_check_atomic_flags() so that will not have to define a strings array for every type of atomic. but that new function will be able to format the error message properly according to atomic type/size

Reply:
technically we can, but such implementation will require complex processing of operation types (op/fetch * combination of atomic ops/fetch-ops/swap/cswap). for now there is single way to check flags & generate messages which is really simple to support (one call + message array), I really feel that this way is better than multiple calls with custom logic.

do you think it should be changed?

Reply:
@yosefe ok, will update :)

Reply:
@yosefe fixed. let me know if previous implementation was better

Reply:
this one is good